### PR TITLE
4.x phpcs

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by ADmad** [comment]
I updated the `phpcs.xml.dist` to include `PSR12` standard (added in phpcs 3.3) and additional useful sniffs from slevomat coding standard. This is a good way to try out the sniffs we want to include in next major release for our `CakePHP` standard.

**Comment by dereuromark** [comment]
Looks good.
There seems to be one missing:
Additional newlines and especially the ones at opening classes etc.

```
class Foo 
{

    // Something

}
```
is not fixed yet to


```
class Foo 
{
    // Something
}
```

as suggested in that link as well as clear from the standard.

File: composer.json [hunk 1]
```diff
@@ -42,7 +42,8 @@
     },
     "require-dev": {
         "phpunit/phpunit": "^6.0",
-        "cakephp/cakephp-codesniffer": "^3.0"
+        "cakephp/cakephp-codesniffer": "^3.0",
+        "slevomat/coding-standard": "^4.6.2"
```
Comment (markstory) [inline composer.json]: I can get this installed in stickler this weekend.

**Comment by ADmad** [comment]
@dereuromark Added sniff for that.

Regarding failing tests:
- It's interesting that tests checking for memory leak have started to fail on PHP 7.2. A bug in PHP?
- The error reported by phpstan is invalid IMO. I have opened an issue for it https://github.com/phpstan/phpstan/issues/1067.

**Comment by ADmad** [comment]
Hmm coverage tests are run when PHP == 7.2. But those memory leak tests are not supposed to run when `CODECOVERAGE` env var is set, so I wonder why they are running.

**Comment by dereuromark** [comment]
Cool! :)

I also opened a phpstan issue today as void issues are not detected.
Had to write a code sniffer here for in the meantime.

File: tests/TestCase/ORM/QueryRegressionTest.php [hunk 2]
```diff
@@ -1605,7 +1605,7 @@ public function testEagerLoadOrderAndSubquery()
     public function testFormatResultsMemoryLeak()
     {
         $this->loadFixtures('Articles', 'Authors', 'Tags', 'ArticlesTags');
-        $this->skipIf(env('CODECOVERAGE') === 1, 'Running coverage this causes this tests to fail sometimes.');
+        $this->skipIf(env('CODECOVERAGE'), 'Running coverage this causes this tests to fail sometimes.');
```
Comment (dereuromark) [inline tests/TestCase/ORM/QueryRegressionTest.php]: Do we need to (bool) cast this env() result?

Comment (ADmad) [reply tests/TestCase/ORM/QueryRegressionTest.php]: No need to get too pedantic in tests :slightly_smiling_face:. Just checking whether the var is defined is enough for deciding whether to skip the test.

Comment (dereuromark) [reply tests/TestCase/ORM/QueryRegressionTest.php]: I am just thinking about skipIf() getter a bool typehint.. :) Then this breaks with any other casted boolish value.

Comment (ADmad) [reply tests/TestCase/ORM/QueryRegressionTest.php]: Done

**Comment by ADmad** [comment]
This is ready to merge.

**Pull request merged**